### ISSUE TITLE
delete doesn't count as turning a property to undefined

### ISSUE THREAD

**Issue by Mathspy** [JSMonk/hegel#161]
```js
function plus1(x) {
    return x + 1;
}
let y: { z: number } = {
    z: 5
};
delete y.z; // if you switch this line to y.z = undefined it will produce an error
let result = plus1(y.z);
```

This should return an error because type `undefined` is not compatible with type `number`

Also lovely work so far!

**Comment by JSMonk** [comment]
Oh, yes. Currently, we don't support `delete`. But it's should be supported. So, we will try to add support of the case. Thank you a lot for your contribution ^_^

**Comment by Mathspy** [comment]
Oh I see! Is there a list of currently unsupported features I can look at before reporting them since they are already known?

**Comment by JSMonk** [comment]
Unfortunately, there is no kind of list. So, if you will report it in issues, it will be great.

**Comment by JSMonk** [comment]
Fixed in 0.0.43 release. Thank you for the contribution ^_^.

**Comment by Mathspy** [comment]
Sweet! Thank you for the amazing work! ❤️
